### PR TITLE
Convert headers to md

### PR DESCRIPTION
**Pull request by TheTripleV**
To be merged after #2690.

### CODE REVIEW

File: source/docs/contributing/frc-docs/style-guide.rst [hunk 1]
```diff
@@ -38,32 +35,27 @@ Use ``.. math::`` for standalone equations and ``:math:`` for inline equations.
...
 
-Interior Whitespace
-^^^^^^^^^^^^^^^^^^^
+### Interior Whitespace
 
 Use one space between sentences.
 
-Headings
---------
+## Headings
```
Comment (sciencewhiz) [inline source/docs/contributing/frc-docs/style-guide.rst:58]: This section should be updated with the md syntax

**Review by sciencewhiz — CHANGES_REQUESTED** [review]

**Comment by sciencewhiz** [comment]
Because of the way it overrides doc8, I want to make sure it works on windows, linux and mac. What did you test it on?

**Comment by TheTripleV** [comment]
I tested on Mac. And ci is Linux. So windows is the only one remaining.

**Comment by TheTripleV** [comment]
Just tried this on windows, make.bat also works

**Review by sciencewhiz — APPROVED** [review]

**Pull request merged**